UnifiedApi/Client/Services/Submitter/Service.cs

```diff
@@ -281,50 +281,65 @@ public string SessionId
 
     while (missing.Count != 0)
     {
-      foreach (var bucket in missing.Batch(10000))
+      foreach (var bucket in missing.Batch(500))
       {
-        try
-        {
-          var partialResults = SessionService.TryGetResults(bucket);
+        var resultStatusCollection = SessionService.GetResultStatus(bucket);
 
-          foreach (var (taskId, bytesArray) in partialResults)
+        foreach (var taskId in resultStatusCollection.IdsReady.Select(tuple => tuple.Item1))
+        {
+          try
           {
             responseHandler(taskId,
-                            bytesArray);
+                            SessionService.TryGetResult(taskId,
+                                                        false,
+                                                        CancellationTokenSource.Token));
+          }
+          catch (Exception e)
+          {
+            errorHandler(taskId,
+                         e.Message + e.StackTrace);
           }
+        }
 
-          results.AddRange(partialResults);
+        resultsCount += resultStatusCollection.IdsReady.Count();
+        missing.ExceptWith(resultStatusCollection.IdsReady.Select(x => x.Item1));
 
-          missing.ExceptWith(partialResults.Select(x => x.Item1));
+        foreach (var taskId in resultStatusCollection.IdsError)
+        {
+          // todo : replace by error from task when reusing tasks ids
+          errorHandler(taskId,
+                       "Task Crashed without result message");
+        }
 
+        missing.ExceptWith(resultStatusCollection.IdsError);
```

Comment:
`IdsError` is the ids that were not returned by the service within `GetResultStatus` Call. Here, you are removing them from the ids that are yet to be fetched.